sktime/base/_base.py

```diff
@@ -14,14 +54,160 @@
+
+        return collected_tags
+
+    @classmethod
+    def get_class_tag(cls, tag_name, tag_value_default=None):
+        """Get tag value from estimator class (only class tags).
+
+        Arguments
+        ---------
+        tag_name : str, name of tag value
+        tag_value_default : any type, default/fallback value if tag is not found
+
+        Returns
+        -------
+        tag_value : value of the tag tag_name in self if found
+                    if tag is not found, returns tag_value_default
+        """
+        collected_tags = cls.get_class_tags()
+
+        if tag_name in collected_tags.keys():
+            return collected_tags[tag_name]
+        else:
+            return tag_value_default
+
+    def get_tags(self):
+        """Get tags from estimator class and dynamic tag overrides.
+
+        Returns
+        -------
+        collected_tags : dictionary of tag names : tag values
+            collected from _tags class attribute via nested inheritance
+            then any overrides and new tags from _tags_dynamic object attribute
+        """
+        collected_tags = type(self).get_class_tags().copy()
+
+        if hasattr(self, "_tags_dynamic"):
+            collected_tags.update(self._tags_dynamic)
+
+        return collected_tags
+
+    def get_tag(self, tag_name, tag_value_default=None):
+        """Get tag value from estimator class and dynamic tag overrides.
+
+        Arguments
+        ---------
+        tag_name : str, name of tag value
+        tag_value_default : any type, default/fallback value if tag is not found
+
+        Returns
+        -------
+        tag_value : value of the tag tag_name in self if found
+                    if tag is not found, returns tag_value_default
+        """
+        collected_tags = self.get_tags()
+
+        if tag_name in collected_tags.keys():
+            return collected_tags[tag_name]
+        else:
+            return tag_value_default
+
+    def set_tags(self, **tag_dict):
+        """Set dynamic tags to given values.
+
+        Arguments
+        ---------
+        tag_dict : dictionary of tag names : tag values
+
+        Returns
+        -------
+        reference to self
+
+        State change
+        ------------
+        sets tag values in tag_dict as dynamic tags in self
+        """
+        self._tags_dynamic.update(tag_dict.copy())
+
+        return self
+
+    def mirror_tags(self, estimator, tag_set=None):
+        """Mirror tags from estimator as dynamic override.
+
+        Arguments
+        ---------
+        estimator : an estimator inheriting from BaseEstimator
+        tag_set : list of str, or str; tag names
+            default = list of all tags in estimator
+
+        Returns
+        -------
+        reference to self
+
+        State change
+        ------------
+        sets tag values in tag_set from estimator as dynamic tags in self
+        """
+        tags_est = estimator.get_tags().copy()
```

Comment:
Why all the copies here? 
```suggestion
        tags_est = estimator.get_tags()
```

Reply:
`dict` are mutable, this is defensive programming to avoid side effects that ultimately could even impact the class dictionary! I.e., an unsuspecting user or extendar could mutate one return dictionary and break tag lookup mechanics for all of sktime.

Reply:
but that should be taken care of in `get_tags`, which it is I believe (we create a new dict every time we call the method)

Reply:
well, but you also want to remove the `copy` there, so you would need at least one.

In any case, I think it's needed in:
* `get_class_tags`
* `get_tags`
* at the start of `set_tags`
* in `clone_tags`

to avoid the various entry points for side effects due to mutability of `dict`

Reply:
inside `get_class_tags` we create a new dict, so I'd say no need for copies, if you're worried about side effects, we should test for them

Reply:
yes, that´s true. But we´d still need the one in `get_tags`, right?

Reply:
resolved in discussion (slightly paranoid use of `deepcopy` it is for now, we can revisit later)